elk/extraction/prompt_collator.py

```diff
@@ -52,12 +52,13 @@ def __init__(
         self.label_fracs = counts / counts.sum()
         if len(self.labels) < 2:
             raise ValueError(f"Dataset {path}/{name} has only one label")
-        if max_examples:
-            self.dataset = self.dataset.select(range(max_examples))
         if dist.is_initialized():
             self.dataset = self.dataset.shard(dist.get_world_size(), dist.get_rank())
 
         self.dataset = self.dataset.shuffle(seed=seed)
+        if max_examples:
+            self.dataset = self.dataset.select(range(max_examples))
```

Comment:
Doesn't that select world_size x max_examples examples?

Reply:
Yep!